src/app/cluster/cluster-details/constraints/component.ts

```diff
@@ -0,0 +1,171 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {ChangeDetectorRef, Component, Input, OnChanges, OnDestroy, OnInit, ViewChild} from '@angular/core';
+import {MatDialog, MatDialogConfig} from '@angular/material/dialog';
+import {MatPaginator} from '@angular/material/paginator';
+import {MatSort} from '@angular/material/sort';
+import {MatTableDataSource} from '@angular/material/table';
+import {OPAService} from '@core/services/opa/service';
+import {UserService} from '@core/services/user/service';
+import {NotificationService} from '@core/services/notification/service';
+import {ConfirmationDialogComponent} from '@shared/components/confirmation-dialog/confirmation-dialog.component';
+import {Cluster} from '@shared/entity/cluster';
+import {Constraint, Kind, Violation} from '@shared/entity/opa';
+import {UserSettings} from '@shared/entity/settings';
+import * as _ from 'lodash';
+import {Subject} from 'rxjs';
+import {take, takeUntil, filter, switchMap} from 'rxjs/operators';
+import {Mode, ConstraintDialog} from './constraint-dialog/component';
+
+@Component({
+  selector: 'km-constraint-list',
+  templateUrl: './template.html',
+  styleUrls: ['./style.scss'],
+})
+export class ConstraintsComponent implements OnInit, OnChanges, OnDestroy {
+  @Input() cluster: Cluster;
+  @Input() projectID: string;
+  @Input() isClusterRunning: boolean;
+  @Input() constraints: Constraint[] = [];
+  settings: UserSettings;
+  dataSource = new MatTableDataSource<Constraint>();
+  displayedColumns: string[] = ['constraintName', 'constraintTemplate', 'match', 'violations', 'actions'];
+  toggledColumns: string[] = ['violationDetails'];
+  isShowDetails = [];
+  @ViewChild(MatSort, {static: true}) sort: MatSort;
+  paginator: MatPaginator;
+  @ViewChild(MatPaginator)
+  set matPaginator(mp: MatPaginator) {
+    this.paginator = mp;
+    if (this.paginator && this.settings) {
+      this.paginator.pageSize = this.settings.itemsPerPage;
+    }
+    this.dataSource.paginator = this.paginator;
+    this._cdr.detectChanges();
+  }
+
+  private _unsubscribe = new Subject<void>();
+
+  constructor(
+    private readonly _opaService: OPAService,
+    private readonly _userService: UserService,
+    private readonly _cdr: ChangeDetectorRef,
+    private readonly _matDialog: MatDialog,
+    private readonly _notificationService: NotificationService
+  ) {}
+
+  ngOnInit(): void {
+    this.dataSource.data = this.constraints;
+    this.dataSource.sort = this.sort;
+    this.dataSource.paginator = this.paginator;
+    this.sort.active = 'constraintName';
+    this.sort.direction = 'asc';
+
+    this._userService.currentUserSettings.pipe(takeUntil(this._unsubscribe)).subscribe(settings => {
+      this.settings = settings;
+    });
+  }
+
+  ngOnChanges(): void {
+    this.dataSource.data = this.constraints;
```

Comment:
I'd actually add a check here if constraints have changed instead of always assigning them. `ngOnChanges` has an argument that provides what has changed.